data/src/main/java/com/android254/data/preferences/DefaultTokenProvider.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 DroidconKE
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android254.data.preferences
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.android254.data.network.util.TokenProvider
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+import java.io.IOException
+import javax.inject.Inject
+
+private object PreferencesKeys {
+    val ACCESS_TOKEN = stringPreferencesKey("access_token")
+}
+
+class DefaultTokenProvider @Inject constructor(private val dataStore: DataStore<Preferences>) :
+    TokenProvider {
+
+    override suspend fun fetch(): Flow<String?> = dataStore.data
+        .catch { exception ->
+            // dataStore.data throws an IOException when an error is encountered when reading data
+            if (exception is IOException) {
+                emit(emptyPreferences())
+            } else {
+                throw exception
```

Comment:
Do we need this?

Reply:
Nope

Thank you. Removed the exception block, and return `emptyPreferences()` if we run into an error

Reply:
@chepsi Final review from you is required.